src/GitHub.App/Services/UsageTracker.cs

```diff
@@ -0,0 +1,376 @@
+            {
+                model.IsEnterpriseUser = true;
+            }
+
+            model.Lang = CultureInfo.InstalledUICulture.IetfLanguageTag;
+            model.AppVersion = AssemblyVersionInformation.Version;
+            model.VSVersion = GitHub.VisualStudio.Services.VisualStudioVersion;
+
+            var counters = new List<IObservable<int>>
+            {
+                GetCounter(GHLaunchCountKeyDay).Do(x => model.NumberOfStartups = x),
+                GetCounter(GHLaunchCountKeyWeek).Do(x => model.NumberOfStartupsWeek = x),
+                GetCounter(GHLaunchCountKeyMonth).Do(x => model.NumberOfStartupsMonth = x),
+                GetCounter(GHCloneCountKey).Do(x => model.NumberOfClones = x),
+                GetCounter(GHCreateCountKey).Do(x => model.NumberOfReposCreated = x),
+                GetCounter(GHPublishCountKey).Do(x => model.NumberOfReposPublished = x),
+                GetCounter(GHCreateGistCountKey).Do(x => model.NumberOfGists = x),
+                GetCounter(GHOpenInGitHubCountKey).Do(x => model.NumberOfOpenInGitHub = x),
+                GetCounter(GHLinkToGitHubCountKey).Do(x => model.NumberOfLinkToGitHub = x),
+                GetCounter(GHLoginCountKey).Do(x => model.NumberOfLogins = x),
+                GetCounter(GHUpstreamPullRequestCount).Do(x => model.NumberOfUpstreamPullRequests = x),
+            };
+
+            if (weekly)
+            {
+                counters.Add(GetCounter(GHLaunchCountKeyWeek)
+                    .Do(x => model.NumberOfStartupsWeek = x));
+            }
+
+            if (monthly)
+            {
+                counters.Add(GetCounter(GHLaunchCountKeyMonth)
+                    .Do(x => model.NumberOfStartupsMonth = x));
+            }
+
+            return Observable.Merge(counters)
+                .ContinueAfter(() => Observable.Return(model));
+        }
+
+        IObservable<Unit> Run()
+        {
+            return Observable.Defer(() =>
+            {
+                // We have a lot of stuff going on at startup so let's wait a little while before we submit our usage stats
+                return Observable.Timer(TimeSpan.FromMinutes(1), TimeSpan.FromHours(8), RxApp.TaskpoolScheduler)
+                    .SelectMany(_ =>
+                        SubmitIfNeeded()
+                            .Catch<Unit, Exception>(ex =>
+                            {
+                                log.Warn("Failed submitting usage data", ex);
+                                return Observable.Return(Unit.Default);
+                            }));
+            });
+        }
+
+        IObservable<int> IncrementCounter(string key)
+        {
+            return Observable.Defer(() =>
+            {
+                try
+                {
+                    // Don't even store data locally if the user opts out.
+                    if (!userSettings.CollectMetrics)
+                    {
+                        return Observable.Empty<int>();
+                    }
```

Comment:
Nitpick Nancy on the prowl: the codebase doesn't use {} when there's only one line.